spec/models/school_spec.rb

```diff
@@ -463,12 +435,104 @@
+    end
+
+    it "does not include participants from other cohorts" do
+      ect = create(:user, :early_career_teacher, school: school)
+      mentor = create(:user, :mentor, school: school)
+
+      expect(school.participants_for(cohort)).not_to include(ect, mentor)
+    end
+
+    it "does not include participants from other schools" do
+      ect = create(:user, :early_career_teacher, cohort: cohort)
+      mentor = create(:user, :mentor, cohort: cohort)
+
+      expect(school.participants_for(cohort)).not_to include(ect, mentor)
+    end
+  end
+
+  describe "#early_career_teacher_profiles_for" do
+    it "includes active ECTs" do
+      ect_profile = create(:early_career_teacher_profile, school: school, cohort: cohort)
+
+      expect(school.early_career_teacher_profiles_for(cohort)).to include ect_profile
+    end
+
+    it "does not include withdrawn ECTs" do
+      ect_profile = create(:early_career_teacher_profile, status: "withdrawn", school: school, cohort: cohort)
+
+      expect(school.early_career_teacher_profiles_for(cohort)).not_to include ect_profile
+    end
+
+    it "does not include ECTs from other cohorts" do
+      ect_profile = create(:early_career_teacher_profile, school: school)
+
+      expect(school.early_career_teacher_profiles_for(cohort)).not_to include ect_profile
+    end
+
+    it "does not include ECTs from other schools" do
+      ect_profile = create(:early_career_teacher_profile, cohort: cohort)
+
+      expect(school.early_career_teacher_profiles_for(cohort)).not_to include ect_profile
+    end
+
+    it "does not include mentors" do
+      mentor_profile = create(:mentor_profile, school: school, cohort: cohort)
+
+      expect(school.early_career_teacher_profiles_for(cohort)).not_to include mentor_profile
+    end
+  end
+
+  describe "#mentor_profiles_for" do
+    it "includes active mentors" do
+      mentor_profile = create(:mentor_profile, school: school, cohort: cohort)
+
+      expect(school.mentor_profiles_for(cohort)).to include mentor_profile
+    end
+
+    it "does not include withdrawn mentors" do
+      mentor_profile = create(:mentor_profile, status: "withdrawn", school: school, cohort: cohort)
+
+      expect(school.mentor_profiles_for(cohort)).not_to include mentor_profile
+    end
+
+    it "does not include mentors from other cohorts" do
+      mentor_profile = create(:mentor_profile, school: school)
+
+      expect(school.mentor_profiles_for(cohort)).not_to include mentor_profile
+    end
+
+    it "does not include mentors from other schools" do
+      mentor_profile = create(:mentor_profile, cohort: cohort)
+
+      expect(school.mentor_profiles_for(cohort)).not_to include mentor_profile
+    end
+
+    it "does not include mentors" do
+      mentor_profile = create(:early_career_teacher_profile, school: school, cohort: cohort)
+
+      expect(school.mentor_profiles_for(cohort)).not_to include mentor_profile
+    end
```

Comment:
Hehe, I think you mean 'it does not include ects' here ;) 

And `ect_profile` for the variable name.

Reply:
Good spot!